packages/sanity/src/form/members/items/ArrayOfObjectsItem.tsx

```diff
@@ -122,8 +142,11 @@ export function ArrayOfObjectsItem(props: MemberItemProps) {
   }, [onPathOpen, member.item.path])
 
   const handleClose = useCallback(() => {
+    if (!member.item.value || isEmpty(member.item.value)) {
+      onRemove()
+    }
     onPathOpen(member.item.path.slice(0, -1))
-  }, [onPathOpen, member.item.path])
+  }, [onPathOpen, member.item.path, member.item.value, onRemove])
```

Comment:
The callback will change every time value changes; I couldnt think of a way to avoid that.

Reply:
Nitpick: It might not be worth it, but we could add a variable named `isEmptyValue` or something to avoid the callback change as often.

```ts
const isEmptyValue = !member.item.value || isEmpty(member.item.value)

const handleClose = useCallback(() => {
  if (isEmptyValue) {
    onRemove()
    onPathOpen(member.item.path.slice(0, -1))
  }
}, [isEmptyValue, member.item.path, onPathOpen])
```

Reply:
Love the idea!